### PR TITLE
[Cider2] use baa as bit-vector library

### PR DESCRIPTION
**Pull request by ekiwi**
This replaces the homespun bit-vector library in Cider with [baa](https://github.com/ekiwi/baa) from crates.io.

### CODE REVIEW

**Review by EclecticGriffin — APPROVED** [review]
This looks great! Very excited to see how this impacts the overall performance. I cleaned up the merge conflict I introduced with my last PR, so this should be good to go.

File: interp/proptest-regressions/tests/values.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Seeds for failure cases proptest has generated in the past. It is
+# automatically read and these particular cases re-run before any
+# novel cases are generated.
+#
+# It is recommended to check this file in to source control so that
+# everyone who runs the test benefits from these saved cases.
+cc 8bfbc0242b84cd41283ecc9c000e88751215cf3bbef619b67f08dcb64f8ab602 # shrinks to input = 0
```
Comment (rachitnigam) [inline interp/proptest-regressions/tests/values.txt:7]: What uses this input file?

Comment (EclecticGriffin) [reply interp/proptest-regressions/tests/values.txt:7]: `cargo test`

Comment (EclecticGriffin) [reply interp/proptest-regressions/tests/values.txt:7]: It's the regression seeds for the `proptest` tests

**Pull request merged**